addon/props/model_props.py

```diff
@@ -86,6 +86,21 @@ class SOURCEOPS_ModelProps(bpy.types.PropertyGroup):
         default=False,
     )
 
+    custom_transform_source: bpy.props.EnumProperty(
+        name='Transform Source',
+        description='Method of specifying $origin and $scale\nEither manually specified in this Panel, or via an object',
+        items=[
+            ('MANUAL', 'Manual Input', 'Specify the Transform manually in this panel', 'NONE', 1),
+            ('CUSTOM_OBJECT', 'Object', 'Use an Object\'s transforms\nIf it is None, then falls back to Manual Input', 'NONE', 2)
+        ]
+    )
+
+    custom_transform_object_ref: bpy.props.PointerProperty(
+        name='Transform Object',
+        description='The Object to use the transform of as the $origin and $scale',
```

Comment:
No need to capitalize the O in Object here IMO.
And speaking of $origin and $scale, what about rotation? I'd use the Z axis rotation.

Reply:
Updated description, should be more correct now.